hw/spinal/config.scala

```diff
@@ -0,0 +1,15 @@
+import spinal.core
+
+package object config {
```

Comment:
I don't think having something at root level is a good thing. Also here it is a specific file, not mixed with user code so I don't think it is a big deal.

Reply:
I do not say that we should have at root level either. What make you think i implied it ?
I mean you can have the object Config in the user package.

I would say that having a package object at root is problematic, for instance if you  include this project as a submodule of another project which does the same, it will conflict.

Reply:
Because if I just remove `package` I get an object at root level. ^^'

Having a `user` package is a really good idea I think. Then in other files it becomes something like `user.config.spinal.generateVerilog(...)`, which looks really pretty!

So I keep the file as `hw/spinal/config.scala`, change `package object config` to `object config` and add `package user` at the beginning?

Or I move the file in `hw/spinal/user/config.scala`? This last path follows more the scala path, but it makes this global file less visible compared to packages for hardware design… :thinking: Maybe name the package `User`?

Reply:
> Because if I just remove package I get an object at root level. ^^'

It isn't what i mean.

> Or I move the file in hw/spinal/user/config.scala? 

Why not 
Having hw/spinal/mylib/Config.scala   ->  object Config{ def sim ... }

> package User

Package name has to be lower case. (convention, amen XD)

Reply:
>  Having hw/spinal/mylib/Config.scala -> object Config{ def sim ... }

Because the goal was to have a global config, not in one library or another